sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -433,3 +433,45 @@ def __str__(self):
         if self._error_format:
             return str(self._error_format)
         return super(ODataV4Error, self).__str__()
+
+class StreamConsumedError(AzureError):
+    """**Provisional** error thrown if you try to access the stream of a response once consumed.
+
+    This error is marked as **provisional**, meaning it may be changed in a future release. It is
+    thrown if you try to read / stream an ~azure.core.rest.HttpResponse or
+    ~azure.core.rest.AsyncHttpResponse once the response's stream has been consumed.
+    """
+    def __init__(self):
+        message = (
+            "You are attempting to read or stream content that has already been streamed. "
+            "You have likely already consumed this stream, so it can not be accessed anymore."
+        )
+        super(StreamConsumedError, self).__init__(message)
+
+class StreamClosedError(AzureError):
+    """**Provisional** error thrown if you try to access the stream of a response once closed.
+
+    This error is marked as **provisional**, meaning it may be changed in a future release. It is
+    thrown if you try to read / stream an ~azure.core.rest.HttpResponse or
+    ~azure.core.rest.AsyncHttpResponse once the response's stream has been closed.
+    """
+    def __init__(self):
+        message = (
+            "The response's content can no longer be read or streamed, since the "
+            "response has already been closed."
+        )
+        super(StreamClosedError, self).__init__(message)
+
+class ResponseNotReadError(AzureError):
+    """**Provisional** error thrown if you try to access a response's content without reading first.
+
+    This error is marked as **provisional**, meaning it may be changed in a future release. It is
+    thrown if you try to access an ~azure.core.rest.HttpResponse or
+    ~azure.core.rest.AsyncHttpResponse's content without first reading the response's bytes in first.
+    """
+
+    def __init__(self):
+        message = (
```

Comment:
From a diagnostics perspective, it may be interesting to know for *which* request this failed. In other words, either provide the response in the constructor, or, at a minimum, take the method + path as arguments to the exception....

Reply:
good point, I'll do something like

"You have not read in the bytes for the response from request '<HttpRequest [GET], url: http://example.com>'. Call .read() on this response first."

Reply:
fixed, thanks!